avocado/core/runner.py

```diff
@@ -395,8 +394,7 @@ def sigtstp_handler(signum, frame):     # pylint: disable=W0613
             self.job.log.debug('')
 
         self.result.check_test(test_state)
-        if not status.mapping[test_state['status']]:
-            failures.append(test_state['name'])
+        summary[test_state['status']] = test_state['fail_class']
```

Comment:
Actually it should also report `ERROR` in case the runner fails badly, which means the job should report JOB_ERROR and not just TEST_FAIL.